File: src/org/jetbrains/haskell/parser/HaskellCompositeElementType.java
package org.jetbrains.haskell.parser;

import com.intellij.psi.tree.IElementType;
import org.jetbrains.haskell.HaskellLanguage;
import org.jetbrains.annotations.NonNls;

/**
 * @author Evgeny.Kurbatsky
 */
public class HaskellCompositeElementType extends IElementType {
    private final String myDebugName;

    public HaskellCompositeElementType(@NonNls String debugName) {
        super(debugName, HaskellLanguage.INSTANCE);
        myDebugName = debugName;
    }

    public String getDebugName() {
        return myDebugName;
    }
}

File: plugin/src/org/jetbrains/haskell/parser/HaskellCompositeElementType.kt
package org.jetbrains.haskell.parser

import com.intellij.psi.tree.IElementType
import org.jetbrains.haskell.HaskellLanguage
import org.jetbrains.annotations.NonNls
import com.intellij.psi.PsiElement
import com.intellij.lang.ASTNode


open class HaskellCompositeElementType(
        debugName: String,
        val constructor : ((ASTNode)->PsiElement)? = null) :
                                IElementType(debugName, HaskellLanguage.INSTANCE) {

    private val myDebugName: String = debugName


    open fun getDebugName(): String {
        return myDebugName
    }
}
